lib/puppet/indirector/rest.rb

```diff
@@ -263,7 +261,9 @@ def do_request(request)
   def handle_response(request, response)
     server_version = response[Puppet::Network::HTTP::HEADER_PUPPET_VERSION]
     if server_version
-      Puppet.push_context({:server_agent_version => server_version})
+      Puppet.lookup(:server_agent_version) do
+        Puppet.push_context({:server_agent_version => server_version})
+      end
```

Comment:
nit, I think you can drop the `{ }`, and just do `push_context(:server_agent_version => server_version)`. Also fine leaving as-is.